cmake/custom/main.cmake

```diff
@@ -35,16 +41,24 @@ add_custom_command(
 add_custom_target(
   binary-info
   ALL
+  COMMAND
+    touch ${PROJECT_SOURCE_DIR}/version.h.in
   DEPENDS
-    ${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}/config.h
+    ${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}/version.h
   )
+
 # See here for the reason why: https://gitlab.kitware.com/cmake/cmake/issues/18399
 set_source_files_properties(${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}/config.h
   PROPERTIES
     GENERATED 1
   )
```

Comment:
These lines are not needed anymore.

Reply:
Only for `config.h`? So for `version.h` it should still be there?

Reply:
Yes, it should be there only for `version.h`. `config.h` is generated with `configure_file` and had the `GENERATED` property implicitly.